lib/compress.js

```diff
@@ -7960,81 +7960,90 @@ merge(Compressor.prototype, {
                     }
                     if (!first.is_boolean(compressor) && !first.is_string(compressor)) break;
                 }
-                return make_node(AST_Array, self, {
-                    elements: self.args
-                });
+                // Array(...) => [ ... ]
+                return make_node(AST_Array, self, { elements: self.args });
               case "Object":
-                if (self.args.length == 0) {
-                    return make_node(AST_Object, self, {
-                        properties: []
-                    });
-                }
+                // Object() => {}
+                if (self.args.length == 0) return make_node(AST_Object, self, { properties: [] });
                 break;
               case "String":
-                if (self.args.length == 0) return make_node(AST_String, self, {
-                    value: ""
-                });
-                if (self.args.length <= 1) return make_node(AST_Binary, self, {
-                    left: self.args[0],
+                // String() => ""
+                if (self.args.length == 0) return make_node(AST_String, self, { value: "" });
+                // String(x) => "" + x
+                if (self.args.length == 1) return make_node(AST_Binary, self, {
                     operator: "+",
-                    right: make_node(AST_String, self, { value: "" })
+                    left: make_node(AST_String, self, { value: "" }),
+                    right: self.args[0],
                 }).optimize(compressor);
                 break;
               case "Number":
-                if (self.args.length == 0) return make_node(AST_Number, self, {
-                    value: 0
-                });
+                // Number() => 0
+                if (self.args.length == 0) return make_node(AST_Number, self, { value: 0 });
+                // Number(x) => +("" + x)
```

Comment:
very minor nitpick - `=>` is an ES6 fat arrow that could be confused for syntax, whereas `--->` can not be.

Reply:
Will have to grep and replace the rest of the file for consistency 😅

Reply:
It's not a big deal either way.  Your call.

Reply:
Done in b8cd933b4dd380c6887a5c7e2e979a4b75a5e904 &minus; should make future text searches easier

Reply:
Cool!

fwiw, only a minifier hacker would find this amusing... prior to your change there were only a few of instances of comments with fat arrows that happened to be valid ES syntax...

```
$ echo 'x ? x : y => x || y' | uglify-js 
x?x:y=>x||y;

$ echo 'x ? y : x => x && y' | uglify-js 
x?y:x=>x&&y;

$ echo 'x ? (a, c) : (b, c) => x ? a : b, c' | uglify-js 
x?(a,c):(b,c)=>x?a:b,c;

$ echo 'x ? y && a : a => (!x || y) && a' | uglify-js 
x?y&&a:a=>(!x||y)&&a;
```

Reply:
Those are the exact lines I glanced upon before making up my mind to grep and replace them 😏